model-generator/src/install_deps.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2023 Robert Bosch GmbH and Microsoft Corporation
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import subprocess
+import sys
+import os
+
+
+def get_script_path() -> str:
+    """Return the absolute path to the directory the invoked Python script is located in."""
+    return os.path.dirname(os.path.realpath(sys.argv[0]))
+
+
+def pip(args: list[str]) -> None:
+    """Invoke the pip process with the given arguments"""
+    subprocess.check_call([sys.executable, "-m", "pip", *args])
+
+
+def should_install_model_generator() -> bool:
+    """Return whether to install the model generator or not."""
+    env = os.getenv("installModelGenerator")
+
+    if env is not None and env:
+        return True
+
+    return False
+
+
+def install_packages():
+    """Install all required Python packages for the model generator and VSpec download."""
+    script_path = get_script_path()
+
+    model_gen_version = "v0.3.0"
+    model_gen_repo = "https://github.com/eclipse-velocitas/vehicle-model-generator.git"
```

Comment:
Don't like that we have dependencies defined here "somewhere" in the code. Maybe we can move this later to a more central place.

Reply:
Moved this configuration to the variables section of the `manifest.json` - now its possible to use the component defaults or overwrite it in a project's `.velocitas.json`